### PR TITLE
Refactor Solo8v2 Vanilla Env to Follow OOP

### LINKED ISSUE #10: Convert the Solo8VanillaEnv to be an ABC and subclass combination
The `Solo8VanillaEnv` was never meant to be an all-in-one package. It's intentioned to be a two-part system, where there is a `BaseEnv` that does a lot of the duplicated work, and the `Solo8VanillaEnv` would just be a subclass of that. Additionally, any new models and/or envs could inherit from `BaseEnv` and it should (in theory) just work like that. 

Unfortunately, when there was only one model, it was easier to debug by putting everything together. This should get resolved before we start heavy design iterations however.

### PR DESCRIPTION
**Pull request by agupta231**
Fixes #10. Basically stripped out as much (forseeable) redundant logic and plopped it into a base environment class `Solo8BaseEnv`.

Thus, if someone wanted to create a new environment, all that they would need to do is to create a new subclass and implement the following methods:
* `load_bodies` - load the robot into the env
* `action_space`
* `reset`
* `step`

..and that's it!

### CODE REVIEW

File: gym_solo/envs/solo8v2vanilla.py [hunk 1]
```diff
@@ -39,37 +39,33 @@ class Solo8VanillaConfig(Solo8BaseConfig):
...
 
 
-class Solo8VanillaEnv(gym.Env):
+class Solo8VanillaEnv(Solo8BaseEnv):
   """An unmodified solo8 gym environment.
   
   Note that the model corresponds to the solo8v2.
   """
   def __init__(self, use_gui: bool = False, realtime: bool = False, 
-               config=None, **kwargs) -> None:
```
Comment (mahajanrevant) [inline gym_solo/envs/solo8v2vanilla.py:48]: Why remove the None?

**Review by mahajanrevant — COMMENTED** [review]
Two small comments

File: gym_solo/envs/solo8_base_env.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+class Solo8BaseEnv(ABC, gym.Env):
+  """Solo 8 abstract base environment."""
+  def __init__(self, config: configs.Solo8BaseConfig, use_gui: bool):
+    """Create a solo8 env.
+
+    Args:
+      config (configs.Solo8BaseConfig): The SoloConfig. Defaults to None.
+      use_gui (bool): Whether or not to show the pybullet GUI. Defaults to 
+        False.
+      realtime (bool): Whether or not to run the simulation in real time.
```
[the remark this answers is not in the document]
Comment (agupta231) [reply gym_solo/envs/solo8_base_env.py:27]: Good catch! I had gotten rid of it :P

File: gym_solo/envs/solo8v2vanilla.py (hunk 1 above)
Comment (agupta231) [reply gym_solo/envs/solo8v2vanilla.py:48]: lol technically returning None is implied so it doesn't really matter... vscode puts it in but I don't when I code--do you have a preference for if we need to keep it or not?

File: gym_solo/envs/solo8_base_env.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (agupta231) [reply gym_solo/envs/solo8_base_env.py:27]: I'll get rid of the comment in the code in the next commit.

**Review by agupta231 — COMMENTED** [review]
Resolved the realtime comment issue, just have a question for you regarding the `-> None`... but that should be in the conversation thread.

**Review by mahajanrevant — APPROVED** [review]
Looks good to me

**Pull request merged**